src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -8,7 +8,23 @@
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidDataException {
+        if (user.getAge() == null || user.getLogin() == null || user.getPassword() == null) {
+            throw new InvalidDataException("All required fields should be completed ");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("this login already exists");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new InvalidDataException("Login is too short");
```

Comment:
The login length check is correctly implemented, but ensure that the minimum length requirement aligns with the task description or typical standards.